### PR TITLE
Store only two rounds

### PR DESCRIPTION
**Pull request by juliangruber**
We only need the current round for preprocessing and the previous round for evaluation. This locks down memory growth of the rounds structure

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -28,10 +32,19 @@ export const startEvaluate = async ({
     cidsSeen.push(cid)
     if (cidsSeen.length > 1000) cidsSeen.shift()
 
+    if (!rounds.current) {
+      rounds.current = new RoundData(roundIndex)
+    } else if (rounds.current.index !== roundIndex) {
+      // This should never happen
+      throw new Error(
+        `Round index mismatch: ${rounds.current.index} !== ${roundIndex}`
+      )
```
Comment (juliangruber) [inline index.js:41]: I've added round validation logic, but am not sure this is worth it. This is just one commit, so easy to revert. Wdyt @bajtos?

Comment (juliangruber) [reply index.js:41]: I guess we should keep it if it's better for a measurement to be rejected than added to the wrong round.

File: lib/round.js [hunk 2]
```diff
@@ -2,7 +2,8 @@ export class RoundData {
   /** @type {Map<string, string>} */
   #knownStrings
 
-  constructor () {
+  constructor (index) {
```
Comment (bajtos) [inline lib/round.js:5]: Would you mind adding a jsdoc comment to describe the index type (is it BigInt?)?

I find the name index a bit ambiguous, roundIndex would be more descriptive, but it also feels a bit redundant to have "round.roundIndex", so I am fine to keep the current name.

File: lib/round.js [hunk 3]
```diff
@@ -2,7 +2,8 @@ export class RoundData {
...
+    this.index = index
```
Comment (bajtos) [inline lib/round.js:6]: AFAICT, we are not using this property anywhere - we always pass the round index in another variable/property alongside the round data. I think we don't need to introduce `this.index`, but it's also possible I have missed something.

WDYT?

File: test/evaluate.js [hunk 4]
```diff
@@ -93,7 +92,7 @@ describe('evaluate', () => {
   })
 
   it('handles empty rounds', async () => {
-    const rounds = { 0: new RoundData() }
+    const round = new RoundData()
```
[the remark this answers is not in the document]
Comment (juliangruber) [reply test/evaluate.js]: Changed it to always pass 0

File: lib/round.js (hunk 3 above)
Comment (juliangruber) [reply lib/round.js:9]: it's used in round index validation logic: https://github.com/filecoin-station/spark-evaluate/pull/135/commits/505ed331b017141ab0a9121ae41fc0b52e903232

Comment (bajtos) [reply lib/round.js:9]: I see. Maybe we don't need to keep the index inside round data, but move it to a new variable in `index.js`, alongside `rounds`, `cidsSeen` and `roundsSeen`?

**Review by bajtos — APPROVED** [review]
LGTM with one more suggestion to consider:
https://github.com/filecoin-station/spark-evaluate/pull/135/files/73c276e5bc42f9f606cb579f6e111961d3f3c545..5253b2e909536abd02be9ba200bbdc3d170b3134#r1463356644

File: lib/round.js (hunk 3 above)
Comment (juliangruber) [reply lib/round.js:9]: What is the motivation for this change?

Comment (juliangruber) [reply lib/round.js:9]: I'm going to merge for now, happy to follow up here

**Pull request merged**